src/swsssdk/configdb.py

```diff
@@ -325,13 +325,138 @@ def get_config(self):
         data = {}
         for key in keys:
             if PY3K:
-                key = key.decode('utf-8')
+                key = key.decode()
             try:
                 (table_name, row) = key.split(self.TABLE_NAME_SEPARATOR, 1)
-                entry = self.__raw_to_typed(client.hgetall(key))
+                entry = self.raw_to_typed(client.hgetall(key))
                 if entry != None:
                     data.setdefault(table_name, {})[self.deserialize_key(row)] = entry
             except ValueError:
                 pass    #Ignore non table-formated redis entries
         return data
 
+
+class ConfigDBPipeConnector(ConfigDBConnector):
+    REDIS_SCAN_BATCH_SIZE = 30
+
+    def __init__(self, **kwargs):
+        super(ConfigDBPipeConnector, self).__init__(**kwargs)
+
+    def __delete_entries(self, client, pipe, pattern, cursor):
+        """Helper method to delete table entries from config db using Redis pipeline
+        with batch size of REDIS_SCAN_BATCH_SIZE.
+        The caller should call pipeline execute once ready 
+        Args:
+            client: Redis client
+            pipe: Redis DB pipe
+            pattern: key pattern
+            cursor: position to start scanning from
+
+        Returns:
+            cur: poition of next item to scan
+        """
+        cur, keys = client.scan(cursor=cursor, match=pattern, count=self.REDIS_SCAN_BATCH_SIZE)
+        for key in keys:
+            pipe.delete(key)
+
+        return cur
+
+    def __delete_table(self, client, pipe, table):
+        """Helper method to delete table entries from config db using Redis pipeline.
+        The caller should call pipeline execute once ready
+        Args:
+            client: Redis client
+            pipe: Redis DB pipe
+            table: Table name.
+        """
+        pattern = '{}{}*'.format(table.upper(), self.TABLE_NAME_SEPARATOR)
+        cur = self.__delete_entries(client, pipe, pattern, 0)
+        while cur != 0:
+            cur = self.__delete_entries(client, pipe, pattern, cur)
+
+    def __mod_entry(self, pipe, table, key, data):
```

Comment:
>__mod_entry [](start = 8, length = 11)

why some functions start with double underscore, but some with one? Suggest use only one.

Reply:
The module is using __ for private and so I maintained it. The function _get_config was an outlier and got fixed by recent push.